app/src/androidTest/java/com/github/HumanLearning2021/HumanLearningApp/firestore/FirestoreDatabaseManagementTest.kt

```diff
@@ -0,0 +1,285 @@
+        assertThat(scratchManagement.putCategory(name), hasName(name))
+    }
+
+    fun test_getCategories() = runBlocking {
+        val cats = demoManagement.getCategories()
+        assertThat(cats, hasItems(hasName("Pomme")))
+    }
+
+    fun test_getAllPictures_categoryNotPresent() = runBlocking {
+        runCatching {
+            scratchManagement.getAllPictures(fakeCategory)
+        }.fold({
+            fail("unexpected successful completion")
+        }, {
+            assertThat(it, instanceOf(IllegalArgumentException::class.java))
+        })
+    }
+
+    fun test_getAllPictures() = runBlocking {
+        val cat = demoManagement.getCategoryById(appleCategoryId)
+        val pics = demoManagement.getAllPictures(cat!!)
+        assertThat(pics.size, equalTo(2))
+        assertThat(pics, hasItems(hasCategory(equalTo(cat))))
+    }
+
+    fun test_removeCategory() = runBlocking {
+        val cat = scratchManagement.putCategory(getRandomString())
+        requireNotNull(scratchManagement.getCategoryById(cat.id), {"category was not put into database"})
+        scratchManagement.removeCategory(cat)
+        assertThat(scratchManagement.getCategoryById(cat.id), equalTo(null))
+    }
+
+    fun test_removePicture() = runBlocking {
+        name = getRandomString()
+        val ctx = ApplicationProvider.getApplicationContext<Context>()
+        val cat = scratchManagement.putCategory(name)
+
+        val tmp = File.createTempFile("meow", ".png")
+        val pic = try {
+            ctx.resources.openRawResource(R.drawable.fork).use { img ->
+                tmp.outputStream().use {
+                    img.copyTo(it)
+                }
+            }
+            val uri = Uri.fromFile(tmp)
+            scratchManagement.putPicture(uri, cat)
+        } finally {
+            tmp.delete()
+        }
+        scratchManagement.removePicture(pic)
+        assertThat(scratchManagement.getPicture(cat), equalTo(null))
+    }
+
+    fun test_putDataset() = runBlocking {
+        val name = getRandomString()
+        val ds = scratchManagement.putDataset(name, setOf())
+        assertThat(scratchManagement.getDatasetById(ds.id)!!.name, equalTo(name))
+    }
+
+    fun test_getDatasetById() = runBlocking {
+        assertThat(demoManagement.getDatasetById("PzuR0B48GpYN5ERxM3DW"), not(equalTo(null)))
+    }
+
+    fun test_getDatasetByName() = runBlocking {
+        assertThat(demoManagement.getDatasetByName("Fruit").size, equalTo(1))
+    }
+
+    fun test_deleteDataset() = runBlocking {
+        val ds = scratchManagement.putDataset(getRandomString(), setOf())
+        requireNotNull(scratchManagement.getDatasetById(ds.id))
+        scratchManagement.deleteDataset(ds.id)
+        assertThat(scratchManagement.getDatasetById(ds.id), equalTo(null))
+    }
+
+    fun test_putRepresentativePicture_categoryNotPresent() = runBlocking {
+        runCatching {
+            scratchManagement.getRepresentativePicture(fakeCategory)
```

Comment:
shouldn't it call **put**RepresentativePicture ?

Reply:
Yes indeed, thank you.